### DISCUSSION TITLE
Get resource path in test

### DISCUSSION THREAD

**Discussion by reidrac** [com-lihaoyi/mill#3586] [Q&A] [answered]
I'm upgrading from mill `0.10.12` to `0.11.12`.

I'm using `getClass.getResource("/").getPath()` in a test to get the path to the test resources, and that was working fine in the old version returning the path to `test/resources/`.

Now with the new version it returns a path in a jar of one of my dependencies (e.g. `/home/user/.cache/coursier/v1/https/repo1.maven.org/maven2/org/slf4j/slf4j-api/2.0.13/slf4j-api-2.0.13.jar!/META-INF/versions/9/`), which I believe is wrong.

My project structure is as follows:
```
├── server
│   ├── resources
│   │   ├── ...
│   ├── src
│   │   └── ...
│   └── test
│       ├── resources
│       │   ├── ...
│       │
│       └── src
│           ├── ...
```

What am I missing?

Any pointers would be very appreciated. Thank you!

**Comment by lihaoyi** [comment, marked as answer]
@reidrac It seems like the the classpath ordering changed which would cause the resource path to resolve differently. In general, you shouldn't rely on classpath ordering, since we can't really guarantee it remains stable over time.

In Mill 0.12.0, we provide a `sys.env("MILL_TEST_RESOURCE_FOLDER")` environment variable that you can use. For 0.11.12, you can emulate it via the following override

```scala
def forkEnv = super.forkEnv() ++
        Map("MILL_TEST_RESOURCE_FOLDER" -> resources().map(_.path).mkString(";")),
```

**Reply by reidrac** [reply]
Thank you, this solved the problem.
